pil_base64_encode_block.py

```diff
@@ -0,0 +1,24 @@
+from base64 import b64encode
+from io import BytesIO
+
+from nio.block.base import Block
+from nio.properties import VersionProperty
+
+
+class PILBase64Encode(Block):
+
+    """ Convert a PIL image to a base64-encoded string """
+
+    version = VersionProperty('0.1.0')
+
+    def __init__(self):
+        super().__init__()
+        self.frame_buffer = BytesIO()
+
+    def process_signals(self, signals, input_id='default'):
+        for signal in signals:
+            signal.image.save(self.frame_buffer, format='JPEG')
```

Comment:
I find it a little annoying that we have to do this all the time in blocks, but you should probably put a `Lock` around this so that the `frame_buffer` doesn't get used by two signal lists at the same time. Debatable if the lock should be inside or outside the for loop. I'm leaning towards outside.

Reply:
you're right, didn't even think of that. If `frame_buffer` were a new object for every list of signals, however, this wouldn't be necessary, right?